client/src/js/references/components/__tests__/Create.test.js

```diff
@@ -0,0 +1,97 @@
+import { CreateReference, mapDispatchToProps } from "../Create";
+
+describe("<CreateReference />", () => {
+    const props = {
+        onSubmit: jest.fn()
+    };
+    let state;
+    const e = {
+        preventDefault: jest.fn(),
+        target: {
+            name: "name",
+            value: "foo",
+            error: "error"
+        }
+    };
+    beforeEach(() => {
+        state = {
+            name: "",
+            description: "",
+            dataType: "genome",
+            organism: "",
+            errorName: "",
+            errorDataType: ""
+        };
+    });
+
+    it("should render", () => {
+        const wrapper = shallow(<CreateReference {...props} />);
+        expect(wrapper).toMatchSnapshot();
+    });
+    it("handleChange should return new state", () => {
+        const wrapper = shallow(<CreateReference {...props} />);
+        wrapper.find("ReferenceForm").simulate("change", e);
+        expect(wrapper.state()).toEqual({
+            ...state,
+            name: "foo"
+        });
+    });
+    it("handleChangeDataType should return dataType", () => {
+        const wrapper = shallow(<CreateReference {...props} />);
+        wrapper.find("DataTypeSelection").simulate("select", "foo");
+        expect(wrapper.state()).toEqual({
+            ...state,
+            dataType: "foo"
+        });
+    });
+    it("handleSubmit should return errorName when [this.state.name.length=0]", () => {
+        const wrapper = shallow(<CreateReference {...props} />);
+        wrapper.setState({ name: [] });
+
+        wrapper.find("form").simulate("submit", e);
+        expect(wrapper.state()).toEqual({
+            ...state,
+            name: [],
+            errorName: "Required Field"
+        });
+    });
+    it("handleSubmit should return errorName when [this.state.dataType.length=0]", () => {
+        const wrapper = shallow(<CreateReference {...props} />);
+        wrapper.setState({ name: [{ foo: "bar" }], dataType: [] });
+        wrapper.find("form").simulate("submit", e);
+        expect(wrapper.state()).toEqual({
+            ...state,
+            dataType: [],
+            name: [{ foo: "bar" }],
+            errorDataType: "Required Field"
+        });
+    });
+    it("handleSubmit should call this.props.onSubmit when [this.state.name.length!=0] and [this.state.dataType.length!=0]", () => {
+        const wrapper = shallow(<CreateReference {...props} />);
+        wrapper.setState({ name: [{ foo: "bar" }], dataType: [{ foo: "bar" }] });
+        wrapper.find("form").simulate("submit", e);
+        expect(props.onSubmit).toHaveBeenCalledWith([{ foo: "bar" }], "", [{ foo: "bar" }], "");
```

Comment:
This test doesn't ensure that the correct parameters are being passed in the right order. You need a different value for each. Also they are supposed to be strings.